frontend/src/context/config/configSlice.tsx

```diff
@@ -41,43 +47,71 @@ export const configSlice = createSlice({
   },
   reducers: {
     updateProjectName: (state, action) => {
-      state.projectName = action.payload
+      state.basic.projectName = action.payload
     },
     updateCalendarType: (state, action) => {
-      state.calendarType = action.payload
+      state.basic.calendarType = action.payload
     },
     updateDateRange: (state, action) => {
       const { startDate, endDate } = action.payload
-      state.dateRange = { startDate, endDate }
+      state.basic.dateRange = { startDate, endDate }
     },
     updateMetrics: (state, action) => {
-      state.metrics = action.payload
+      const {
+        VELOCITY,
+        CYCLE_TIME,
+        CLASSIFICATION,
+        LEAD_TIME_FOR_CHANGES,
+        DEPLOYMENT_FREQUENCY,
+        CHANGE_FAILURE_RATE,
+        MEAN_TIME_TO_RECOVERY,
+      } = REQUIRED_DATA
+
+      state.basic.metrics = action.payload
+
+      state.isShowBoard = [VELOCITY, CYCLE_TIME, CLASSIFICATION].some((metric) => state.basic.metrics.includes(metric))
+      state.isShowPipeline = [
+        LEAD_TIME_FOR_CHANGES,
+        DEPLOYMENT_FREQUENCY,
+        CHANGE_FAILURE_RATE,
+        MEAN_TIME_TO_RECOVERY,
+      ].some((metric) => state.basic.metrics.includes(metric))
+      state.isShowSourceControl = [LEAD_TIME_FOR_CHANGES].some((metric) => state.basic.metrics.includes(metric))
     },
+    updateBasicConfigState: (state, action) => {
+      state.basic = action.payload
+      state.boardConfig = action.payload.board || state.boardConfig
+      state.pipelineToolConfig = action.payload.pipelineTool || state.pipelineToolConfig
+      state.sourceControlConfig = action.payload.sourceControl || state.sourceControlConfig
+    },
```

Comment:
shall we keep the state name are same? 
boardConfig || board
pipelineToolConfig || pipelineTool
sourceControlConfig || sourceControl
as there should be used to same component.

Reply:
yes, unit our name is better choice!